### PR TITLE
Add reportType=native to surface diffs in Gradle's test report

### LINKED ISSUE #2263: Support attaching snapshot images to Gradle test results
https://docs.gradle.org/9.4.0-rc-1/release-notes.html#additional-test-data-capture

This new feature seems relevant and useful to paparazzi:

Image

### PR DESCRIPTION
**Pull request by jrodbx**
Introduces the `app.cash.paparazzi.reportType` Gradle property. It defaults to `legacy`, which is today's behavior unchanged: Paparazzi replaces Gradle's test reporter with its own and embeds diff images in the generated page.

Setting it to `native` leaves Gradle's report generator in place and instead publishes each snapshot diff as a FileEntry attachment on the test that produced it, so diffs appear in the attachments tab of Gradle's own report. This requires Gradle 9.4 or later, enforced at plugin-apply time.

Native mode routes JUnit 4 tests through a wrapping JUnit Platform engine, published as a new paparazzi-junit-platform artifact so that JUnit Platform stays out of the always-loaded paparazzi runtime. The two sides share no typed link: the engine discovers diffs by filename convention from paparazzi.failures.dir at executionFinished. The plugin adds the artifact to the test runtime classpath and excludes the real junit-vintage engine, which would otherwise discover the same tests a second time.

Not installing a custom TestReporter also keeps native mode clear of the Gradle internal reporting APIs that have repeatedly broken across releases (#1745, #2182, #2111).

Accessibility artifacts are not migrated yet. AccessibilityRenderExtension still bakes its overlay into the snapshot in both modes; publishing those as attachments is the follow-up that lets legacy be removed.

Obsoletes https://github.com/cashapp/paparazzi/pull/2299.  Closes https://github.com/cashapp/paparazzi/issues/2263.

### CODE REVIEW

**Review by geoff-powell — APPROVED** [review]

File: paparazzi-junit-platform/src/main/java/app/cash/paparazzi/junitplatform/PaparazziExecutionListener.kt [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    val prefix = "delta-${packageName}_${simpleClassName}_$methodName"
+
+    failureDir.listFiles()
+      ?.filter { file ->
+        val name = file.name
+        if (!name.startsWith(prefix) || !name.endsWith(".png")) return@filter false
+        // Ensure prefix is followed by '.' (no label) or '_' (label suffix) so
+        // a method like `testThing` doesn't accidentally match `testThingExtra`.
+        val boundary = name.getOrNull(prefix.length)
+        boundary == '.' || boundary == '_'
```
Comment (colinmarsch) [inline paparazzi-junit-platform/src/main/java/app/cash/paparazzi/junitplatform/PaparazziExecutionListener.kt:91]: Not sure I understand these boundary values fully? `_` is valid in test method names and `[` is used in parameterized tests but seem to be rejected?

File: paparazzi-gradle-plugin/src/test/java/app/cash/paparazzi/gradle/PaparazziPluginTest.kt [hunk 2]
```diff
@@ -794,6 +794,34 @@ class PaparazziPluginTest {
...
+
+    val deltaName = "delta-app.cash.paparazzi.plugin.test_VerifyTest_verify.png"
+    val delta = File(fixtureRoot, "build/paparazzi/failures/debug/$deltaName")
+    assertThat(delta.exists()).isTrue()
+
+    // Assert against the rendered HTML, not the binary event log: that log also contains
+    // the failure message's copy of the delta path, so matching it would pass either way.
+    val testReportDir = File(fixtureRoot, "build/reports/tests/testDebugUnitTest")
+    val attachmentReferenced = testReportDir.walkTopDown()
+      .any { it.isFile && it.extension == "html" && it.readText().contains(deltaName) }
```
Comment (colinmarsch) [inline paparazzi-gradle-plugin/src/test/java/app/cash/paparazzi/gradle/PaparazziPluginTest.kt:821]: Should we check more specifically here that this is in the attachments or is an ``? As is this could pass if the filename appears in failure text.

**Review by colinmarsch — APPROVED** [review]
This looks good to me! Just left a couple comments

File: paparazzi-junit-platform/src/main/java/app/cash/paparazzi/junitplatform/PaparazziExecutionListener.kt (hunk 1 above)
Comment (jrodbx) [reply paparazzi-junit-platform/src/main/java/app/cash/paparazzi/junitplatform/PaparazziExecutionListener.kt]: Good catch on `[` (confirmed with a test). 

Parameterized writes `delta-…_verify[0].png`, but Vintage's `MethodSource` reports the bare verify. 

Before: delta on disk, no attachment. 
After: it's in the attachments tab.

See test `verifyFailureNativeReportParameterized`.

Re: `_`, I kept as-is.  `…_verify_baseline_accessibility.png` is identical for method `verify baseline accessibility` and for `verify baseline` with label accessibility, since `toFileName` uses `_` for both. Over-matching beats silently dropping an image; something to fix in the future.

File: paparazzi-gradle-plugin/src/test/java/app/cash/paparazzi/gradle/PaparazziPluginTest.kt (hunk 2 above)
Comment (jrodbx) [reply paparazzi-gradle-plugin/src/test/java/app/cash/paparazzi/gradle/PaparazziPluginTest.kt]: Fixed.  Now matching `<img src="…delta-….png"`, which only the attachments markup produces.

**Comment by jrodbx** [comment]
odd, this auto-merged despite a failed Windows CI build and I was relying on that to add/fix some stuff.  fast follow incoming.

**Comment by colinmarsch** [comment]
> odd, this auto-merged despite a failed Windows CI build and I was relying on that to add/fix some stuff. fast follow incoming.

Ah, I saw that failure and it looked like a dep download flake so I re-ran it and I guess it passed.
